### PR TITLE
feat(permissions): Add ability to specify permissions in a file, rather

### PR DESCRIPTION
**Pull request by ttomsu**
than some external provider.

### CODE REVIEW

File: fiat-file/src/main/java/com/netflix/spinnaker/fiat/roles/file/FileBasedUserRolesProvider.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+@Slf4j
+@Component
+@ConditionalOnProperty(value = "auth.groupMembership.service", havingValue = "file")
+public class FileBasedUserRolesProvider implements UserRolesProvider {
+
+  @Autowired
+  ConfigProps configProps;
+
+  private Map<String, List<Role>> parse() throws IOException {
+    return parse(new BufferedReader(new FileReader(new File(configProps.getPath()))));
```
Comment (jtk54) [inline fiat-file/src/main/java/com/netflix/spinnaker/fiat/roles/file/FileBasedUserRolesProvider.java]: Thanks, Java.

File: fiat-file/src/main/java/com/netflix/spinnaker/fiat/roles/file/FileBasedUserRolesProvider.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    try {
+      return new ArrayList<>(parse().get(userId));
+    } catch (IOException io) {
+      log.error("Couldn't load roles for user " + userId + " from file", io);
+    }
+    return Collections.emptyList();
+  }
+
+  @Override
+  public Map<String, Collection<Role>> multiLoadRoles(Collection<String> userIds) {
```
Comment (jtk54) [inline fiat-file/src/main/java/com/netflix/spinnaker/fiat/roles/file/FileBasedUserRolesProvider.java]: Maybe add some logging for `load` and `multiLoad` for some traceability.

Comment (ttomsu) [inline fiat-file/src/main/java/com/netflix/spinnaker/fiat/roles/file/FileBasedUserRolesProvider.java]: The caller prints out the results at the `debug` level, so additional logging here would be duplicate

Comment (jtk54) [inline fiat-file/src/main/java/com/netflix/spinnaker/fiat/roles/file/FileBasedUserRolesProvider.java]: Got it

**Pull request merged**